pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -434,6 +435,15 @@ func newKubeVirtConfigForCR(cr *hcov1alpha1.HyperConverged, namespace string) *c
 			"selinuxLauncherType": "spc_t",
 		},
 	}
+	val, ok := os.LookupEnv("SMBIOS")
+	if ok && val != "" {
+		cm.Data[virtconfig.SmbiosConfigKey] = val
+	}
+	val, ok = os.LookupEnv("MACHINETYPE")
+	if ok && val != "" {
+		cm.Data[virtconfig.MachineTypeKey] = val
```

Comment:
1. IIUC, we are putting nested key-value maps `SMBIOS` and `MACHINETYPE` inside the kubevirt-config CM? It means the consumer must follow that.

2. The jira ticket mentioned also an SKU version (which is the kubevirt version). Where is this referenced? inside SMBIOS internal key-value map?

Reply:
> 1. IIUC, we are putting nested key-value maps `SMBIOS` and `MACHINETYPE` inside the kubevirt-config CM? It means the consumer must follow that.

kubevirt expects a json or yaml serialized string, the user should put there what kubevirt expects.
https://github.com/kubevirt/kubevirt/blob/eaa5d58a86523ee4489e369f043eeb2a1ac7ed3b/pkg/virt-config/config-map.go#L291

in hack/build-manifests.sh we are injecting a small piece of yaml

> 2. The jira ticket mentioned also an SKU version (which is the kubevirt version). Where is this referenced? inside SMBIOS internal key-value map?

Yes, the value of Product key in smbios map